roxie/ccd/ccdquery.cpp

```diff
@@ -515,6 +524,9 @@ class CQueryFactory : public CInterface, implements IQueryFactory, implements IR
         if (!helperFactory)
             throw MakeStringException(ROXIE_INTERNAL_ERROR, "Internal error: helper function %s not exported", helperName.str());
 
+        RemoteActivityId remoteId(rid, hashValue);
+        RemoteActivityId remoteId2(rid | ROXIE_ACTIVITY_FETCH, hashValue);
+
```

Comment:
remoteId id used in most of the case statements below so it makes sense to make the call all the time, but remoteId2 is only use in the 1 case for TAKkeyedjoin, TAKkeyeddenormalize, TAKkeyeddenormalizegroup - should we always create it?